src/manifest.js

```diff
@@ -78,6 +78,15 @@ export const parseManifest = ({
       });
     }
   }
+  if (manifest.segments && manifest.segments.length && !manifest.targetDuration) {
+    const lastSegment = manifest.segments[manifest.segments.length - 1];
+    const lastDuration = lastSegment && Math.round(lastSegment) || 10;
+
+    if (onwarn) {
+      onwarn(`manifest has no targetDuration defaulting to ${lastDuration}`);
```

Comment:
I think we may be better off using the largest segment duration found (and 10 if no segments included), since the TARGETDURATION is supposed to be the highest duration that will be encountered in a stream (and we will run into issues if the target duration is shorter than some segment durations).

Would also be worth having a test for it.